src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Windows/CertificatePal.cs

```diff
@@ -490,10 +510,12 @@ public void Dispose()
             _certContext = null;
             if (certContext != null && !certContext.IsInvalid)
             {
-                if (!_certContextCloned)
+                if (_certContextCloned)
                 {
-                    certContext.Dispose();
+                    certContext.DangerousRelease();
                 }
+
+                certContext.Dispose();
```

Comment:
Note, certContext.Dispose does not result in a double (or more) release (safehandle protects against that)